scripts/2.TCGA-MLexample-covariates.py

```diff
@@ -90,55 +91,64 @@
 # Typically, this can only be done where the number of mutations is large enough
 X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=0)
 
-# Select the feature set for the different models
-N_COVARIATES = covariates.shape[1]
+
+# ## Feature selection
+
+# In[11]:
+
+# Select the feature set for the different models within the pipeline
+N_COVARIATES = len(covariates.columns)
 def select_feature_set_columns(X, feature_set):
     if feature_set=='expressions': return X[:, N_COVARIATES:]
     return  X[:, :N_COVARIATES]
 
+expression_features = Pipeline([
+    ('select_features', FunctionTransformer(select_feature_set_columns,
+        kw_args={'feature_set': 'expressions'})),
+    ('standardize', StandardScaler()),
+    ('pca', PCA())
+])
+covariate_features = Pipeline([
+    ('select_features', FunctionTransformer(select_feature_set_columns,
+        kw_args={'feature_set': 'covariates'})),
+    ('standardize', StandardScaler())
+])
 
-# ## Define pipeline and Cross validation model fitting
 
-# In[11]:
+# ## Elastic net classifier and model paraemeters
+
+# In[12]:
 
 # Parameter Sweep for Hyperparameters
 n_components_list = [50, 100]
+regularization_alpha_list = [10 ** x for x in range(-3, 1)]
+regularization_l1_ratio = 0.15
 
 param_grids = {
     'full': {
         'features__expressions__pca__n_components' : n_components_list,
-        'classify__loss': ['log'],
-        'classify__penalty': ['elasticnet'],
-        'classify__alpha': [10 ** x for x in range(-3, 1)],
-        'classify__l1_ratio': [0, 0.2, 0.8, 1],
+        'classify__alpha': regularization_alpha_list
     },
     'expressions': {
         'features__expressions__pca__n_components' : n_components_list,
-        'classify__loss': ['log'],
-        'classify__penalty': ['elasticnet'],
-        'classify__alpha': [10 ** x for x in range(-3, 1)],
-        'classify__l1_ratio': [0, 0.2, 0.8, 1],
+        'classify__alpha': regularization_alpha_list
     },
     'covariates': {
```

Comment:
Does it break the grid search to add a step that isn't in the pipeline? If not, the same `param_grid` could be used for all three models?